strawberry/schema/schema.py

```diff
@@ -78,7 +80,12 @@ def __init__(
 
         graphql_types = []
         for type_ in types:
-            graphql_type = self.schema_converter.from_object(type_._type_definition)
+            graphql_type = self.schema_converter.from_maybe_optional(type_)
+            if isinstance(graphql_type, GraphQLNonNull):
+                graphql_type = graphql_type.of_type
+            assert isinstance(
+                graphql_type, GraphQLNamedType
+            ), f"{graphql_type} is not a named GraphQL Type"
```

Comment:
I see that @patrick91 requested this be an `assert` but I think this would just be better as a custom exception type. Asserts shouldn't typically be used to check logic the user can influence (except in tests).

Reply:
yes, you're right :) a custom exception is better 😊 sorry @paulo-raca!

Reply:
Can it be a ValueError? :thinking:

Reply:
let's make our own custom exception :)

`InvalidTypeError` or something like this :)

Reply:
`SchemaConverter.from_type()` raises TypeError. Sounds close enough :)